apps/server/src/modules/task/uc/task.uc.ts

```diff
@@ -163,6 +164,39 @@ export class TaskUC {
 		return [taskWithStatusVos, total];
 	}
 
+	// it should return also the scopePermissions for this user added to the entity .scopePermission: { userId, read: boolean, write: boolean }
+	// then we can pass and allow only scoped courses to getPermittedLessonIds and validate read write of .scopePermission
+	private async getPermittedCourses(user: User, neededPermission: Actions): Promise<Course[]> {
+		let permittedCourses: Course[] = [];
+
+		if (neededPermission === Actions.write) {
+			[permittedCourses] = await this.courseRepo.findAllForTeacher(user.id);
```

Comment:
We should look in the next iteration on it. 
On this place we must know which request is mapped to the specifc permission. 
Maybe we can make a jump repo method for getWriteCourses that call findAllForTeachers in course repo.

Same for lesson with the explicit flag hidden: false that can be hide behind the request. 
The only point is that we move "Domain" permissions behind repo methods.

But not sure. Let keep it in mind for later.